pkg/types/validation/installconfig.go

```diff
@@ -998,8 +998,14 @@ func validateFeatureSet(c *types.InstallConfig) field.ErrorList {
 	if c.FeatureSet != configv1.TechPreviewNoUpgrade {
 		errMsg := "the TechPreviewNoUpgrade feature set must be enabled to use this field"
 
-		if c.GCP != nil && len(c.GCP.NetworkProjectID) > 0 {
-			allErrs = append(allErrs, field.Forbidden(field.NewPath("platform", "gcp", "networkProjectID"), errMsg))
+		if c.GCP != nil {
+			if len(c.GCP.NetworkProjectID) > 0 {
+				allErrs = append(allErrs, field.Forbidden(field.NewPath("platform", "gcp", "networkProjectID"), errMsg))
+			}
+
+			if len(c.GCP.CreateFirewallRules) > 0 && c.GCP.CreateFirewallRules != gcp.CreateFirewallRulesEnabled {
```

Comment:
```suggestion
			if len(c.GCP.CreateFirewallRules) > 0 {
```
nit: Any usage of this field, even when it's consistent with the existing functionality, should be protected by the flag. The main concern is we want the freedom to change the field, so we want users to be warned that we may break them in the future.